php/elFinderVolumeGoogleDrive.class.php

```
@@ -0,0 +1,1659 @@
+				array_merge($uploadname,$name);
+				$files->setName($name);
+				$files->setDescription('');
+				$stat['mime'] == '' ? $mimeType= parent::$mimetypes[pathinfo(basename($path),PATHINFO_EXTENSION)] : $mimeType= $stat['mime'];			
+				$files->setMimeType($mimeType);
+							
+				//Set the Folder Parent			
+				basename(dirname($path)) == '' ? $parentId = 'root' : $parentId = basename(dirname($path));	
+				
+				$files->setParents(array($parentId));
+				
+				$data= stream_get_contents($fp);
+				$file = $this->oauth->files->create($files, array(
+					  'data' => strlen($data) == 0 ? ' ' : $data,
+					  'mimeType' => $mimeType,
+					  'uploadType' => 'media'
+				));			
+		  }			
+		} catch (Exception $e) {
+			return $this->setError('GoogleDrive error: '.$e->getMessage());
+		}
+				
+		return $path;
+	}
+
+	/**
+	 * Get file contents
+	 *
+	 * @param  string  $path  file path
+	 * @return string|false
+	 * @author Dmitry (dio) Levashov
+	 **/
+	protected function _getContents($path) {	
+		$contents = '';
+		
+		try {
+			//$files = new Google_Service_Drive_DriveFile();						
+			$itemId = basename($path);
+			
+			$contents = $this->oauth->files->get($itemId, [
+				'alt' => 'media'
+			]);
+			$contents = (string) $contents->getBody();     			
+		} catch (Exception $e) {
+			return $this->setError('GoogleDrive error: '.$e->getMessage());
+		}
+		
+		return $contents;
+	}
+
+	/**
+	 * Write a string to a file
+	 *
+	 * @param  string  $path     file path
+	 * @param  string  $content  new file content
+	 * @return bool
+	 * @author Dmitry (dio) Levashov
+	 **/
+	protected function _filePutContents($path, $content) {	
+		$res = false;
+		
+		if ($local = $this->getTempFile($path)) {
+			if (@file_put_contents($local, $content, LOCK_EX) !== false
+			&& ($fp = @fopen($local, 'rb'))) {
+				clearstatcache();
+				$res = $this->_save($fp, $path, '', array());
+				@fclose($fp);
+			}
+			file_exists($local) && @unlink($local);
```

Do not use the `@` to hide the error.